### ISSUE TITLE
Output panel: method body after : is considered return type

### ISSUE THREAD

**Issue by Gama11** [fdorg/flashdevelop#1061]
For example, a function with no return type, only a `:`, leads to the output panel displaying the entire method body as the return type of the method. This is a bit annoying / confusing while typing out code.

![](http://i.imgur.com/4cCP3Sj.png)

I think it's possible for the parser to tell that this isn't a `typedef ` return.
I'm fairly confindent this didn't used to happen, however it does happen on the master branch.

**Comment by Neverbirth** [comment]
You sure this worked before? As you say, it's already broken in the latest official release, but If this ever worked right, by looking at the repository history, it was like 2 years ago, so...

Yes, the parser could handle it, when we are in a type declaration and we have at least one bracket, if we find symbols like .;, or ( when not in a subtype, we can "stop" the parser. But it would be nice to see the Haxe parser logic completely separate from the other languages so it can be made cleaner, and I also have some additional ideas for the future.

**Comment by elsassph** [comment]
I think your recent parser changes made the support for "anonymous" types a lot more loose which creates this situation. 
That's why originally I was cancelling the type when finding `;` for instance.
Another possibility would be to drop the type if there's a newline between `:` and the first `{`.

**Comment by Neverbirth** [comment]
No, this already happened before my changes.

**Comment by Neverbirth** [comment]
And lastly, I only changed ; when in generics, because you can have them in there in Haxe, but admittedly only in generic constraints, not any generic case.

**Comment by elsassph** [comment]
Ah ok.
Yes it is technically valid syntax, but that's a syntax error we need to recover from it somehow.

**Comment by Neverbirth** [comment]
Yes, I agree completely, but using new line to stop the parser wouldn't be the proper way, we add support for a wrong syntax case but break many valid syntax cases :P.

I'd love to see that parser separation, it will be helpful for everyone and will ease adding new cases.

**Comment by elsassph** [comment]
We're talking mostly about the outline view, it wouldn't break so many cases in practice. Plus I really don't see how it could be really used in real life.

**Comment by Neverbirth** [comment]
I'd say it's more serious than that. Of course it's not immediately visible because of the compiler service, but It means the model we have in memory is also wrong, so who knows how it may affect in some cases like refactoring, code generation, etc.

But yes, I'd mark this as low priority.

**Comment by elsassph** [comment]
I'd say it's more problematic to use the function body as anonymous :)
